### PR TITLE
Upgrade AllenNLP to `v2.2.0`

### PR DESCRIPTION
**Pull request by himkt**
This PR re-introduces AllenNLP to the dependencies.
(AllenNLP v2.2.0 supports PyTorch v1.8.0)

## Motivation
- restore AllenNLP in `setup.py`
- remove specific test configuration for AllenNLP

## Description of the changes
- 4a91efc updates `setup.py`
- 88311e9 changes GitHub Actions

(~verifying on https://github.com/himkt/optuna/pull/13, after the examples finish successfully, I'll remove WIP~ => done!)

### CODE REVIEW

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -57,8 +57,7 @@ jobs:
       env:
         OMP_NUM_THREADS: 1
       run: |
-        pytest --cov=optuna --cov-report=xml tests \
-            --ignore tests/integration_tests/allennlp_tests
+        pytest --cov=optuna --cov-report=xml tests
```
Comment (crcrpar) [inline .github/workflows/coverage.yml:60]: 👍

**Review by crcrpar — APPROVED** [review]

File: .github/workflows/tests-integration.yml [hunk 2]
```diff
@@ -65,18 +65,9 @@ jobs:
             --ignore tests/integration_tests/allennlp_tests \
```
Comment (toshihikoyanase) [inline .github/workflows/tests-integration.yml:65]: AllenNLP seems to support for Python 3.6. Do we need to exclude the AllenNLP related tests here?
https://pypi.org/project/allennlp/

> Requires: Python >=3.6.1

```suggestion
```

File: .github/workflows/examples.yml [hunk 3]
```diff
@@ -53,9 +53,9 @@ jobs:
         if [ ${{ matrix.python-version }} = 3.6 ]; then
           IGNORES='chainermn_.*|fastai*|pytorch_distributed_.*|rapids_.*|botorch_.*|pytorch/pytorch_checkpoint*|allennlp*'
```
Comment (toshihikoyanase) [inline .github/workflows/examples.yml:54]: ditto.
AllenNLP was excluded by #2442 when PyTorch==1.8.0 was introduced. I think we can activate the AllenNLP related tests in Python 3.6 now.

```suggestion
          IGNORES='chainermn_.*|fastai*|pytorch_distributed_.*|rapids_.*|botorch_.*|pytorch/pytorch_checkpoint*'
```

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your PR.
The change basically looks good to me, but I added comments on Python 3.6.

**Comment by himkt** [comment]
@toshihikoyanase Thank you for the catch. :bow: Revised the PR.

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your update. LGTM!

**Pull request merged**